dsc/workflows/base/__init__.py

```diff
@@ -93,97 +91,30 @@ def _get_subclasses(cls) -> Iterator[type[Workflow]]:
             yield from subclass._get_subclasses()  # noqa: SLF001
             yield subclass
 
-    def reconcile_bitstreams_and_metadata(self) -> tuple[set[str], set[str]]:
+    @abstractmethod
+    def reconcile_bitstreams_and_metadata(self) -> None:
         """Reconcile bitstreams against metadata.
 
-        Generate a list of bitstreams without item identifiers and item identifiers
-        without bitstreams. Any discrepancies will be addressed by the engineer and
-        stakeholders as necessary.
-        """
-        bitstream_dict = self._build_bitstream_dict()
-
-        # extract item identifiers from batch metadata
-        item_identifiers = [
-            self.get_item_identifier(item_metadata)
-            for item_metadata in self.item_metadata_iter()
-        ]
-
-        # reconcile item identifiers against bitstreams
-        item_identifiers_with_bitstreams = self._match_item_identifiers_to_bitstreams(
-            bitstream_dict.keys(), item_identifiers
-        )
-
-        bitstreams_with_item_identifiers = self._match_bitstreams_to_item_identifiers(
-            bitstream_dict.keys(), item_identifiers
-        )
-
-        logger.info(
-            "Item identifiers from batch metadata with matching bitstreams: "
-            f"{item_identifiers_with_bitstreams}"
-        )
-
-        item_identifiers_without_bitstreams = set(item_identifiers) - set(
-            item_identifiers_with_bitstreams
-        )
-        bitstreams_without_item_identifiers = set(bitstream_dict.keys()) - set(
-            bitstreams_with_item_identifiers
-        )
+        Items in DSpace represent a "work" and combine metadata and files,
+        known as "bitstreams". For any given workflow, the purpose of this
+        method is to determine whether sufficient data has been provided
+        for a batch of items to be submitted into DSpace, but how this
+        method accomplishes this varies based on the data provided
+        to execute the workflow.
 
-        return item_identifiers_without_bitstreams, bitstreams_without_item_identifiers
+        This method ensures the existence of both bitstreams and metadata
+        for each item in the batch, verifying that all provided bitstreams
+        can be linked to a metadata record and vice versa.
```

Comment:
What about _combining_ the first two paragraphs from this docstring:

```text
Items in DSpace represent a "work" and combine metadata and files,
known as "bitstreams". For any given workflow, this method ensures 
the existence of both bitstreams and metadata for each item in the 
batch, verifying that all provided bitstreams can be linked to a 
metadata record and vice versa.
```

Felt like it was superbly articulated, just a little redundant.

Reply:
Agreed!

Reply:
I wholeheartedly agree! Change has been made.